api_server/src/request/drive.rs

```diff
@@ -74,222 +81,213 @@ impl PatchDrivePayload {
     }
 }
 
-impl IntoParsedRequest for PatchDrivePayload {
-    fn into_parsed_request(
-        self,
-        id_from_path: Option<String>,
-        method: Method,
-    ) -> result::Result<ParsedRequest, String> {
-        match method {
-            Method::Patch => {
-                self.validate()?;
-                let drive_id: String = self.get_string_field_unchecked("drive_id");
-                let path_on_host: String = self.get_string_field_unchecked("path_on_host");
-
-                let id_from_path = id_from_path.unwrap_or_default();
-                if id_from_path != drive_id {
-                    return Err(String::from(
-                        "The id from the path does not match the id from the body!",
-                    ));
-                }
+pub fn parse_put_drive(
+    maybe_body: Option<&Body>,
+    id_from_path: Option<&&str>,
+) -> Result<ParsedRequest, Error> {
+    METRICS.put_api_requests.drive_count.inc();
+    let id = match id_from_path {
+        Some(&id) => checked_id(id)?,
+        None => {
+            return Err(Error::EmptyID);
+        }
+    };
 
-                let (sender, receiver) = oneshot::channel();
-                Ok(ParsedRequest::Sync(
-                    VmmRequest::new(
-                        VmmAction::UpdateBlockDevicePath(drive_id, path_on_host),
-                        sender,
-                    ),
-                    receiver,
-                ))
-            }
-            _ => Err(format!("Invalid method {}!", method)),
+    if let Some(body) = maybe_body {
```

Comment:
Most requests do this check against `Option<&Body>` and all return the same error on `None`.

Maybe add a static function to de-duplicate this code.

Reply:
I refactored that part and now there is more consistency. Please see if it fits the larger context.